### ISSUE TITLE
"Note to self" weird behaviour with dual SIM phone

### ISSUE THREAD

**Issue by ijackson** [signalapp/Signal-Android#12265]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Describe here the issue that you are experiencing.

### Steps to reproduce

 - On a dual sim phone with Signal set up and registered with one of the two numbers, and with Signal as default SMS app
 - Let's call the Signal-registered number the "main" number and the other one the "aux" number
 - Open Signal
 - From the main conversations screen, tap "write" to start a new message
 - Enter my aux number into the input field in `+44` format.
 - My own "contact" entry for that number comes up.  Tap it.
 - Now I am looking at my conversation "with" my own aux number
 - Hold down the `+` button to bring up the delivery options
 - Select my own *main* SIM.  Now I am writing an SMS to be sent via my main SIM, to the number for aux SIM.
 - Enter some text and hit send.

**Actual result:** The message appears as outgoing in my conversation "with" my aux number.  But nothing else happens.  In particular, in my conversation "with" my own main number (which Signal titles "Note to self") the message is not visible.  I haven't been able to verify, but I don't think any SMS was actually sent.

**Expected result:** I should get a notification about an incoming message.  The message should appear in Signal in my "Note to self", and show as having arrived via my aux SIM.

The SMS flow in the other direction operates normally: if I go to "Note to Sefl", use the message type selector to select sending SMS using my aux SIM, and send a message, the message generates a notifcation and is delivered and displayed as expected (in my conversation "with" my aux number).

I think something is suppressing messages in "Note to self".  I think this is probably appropriate for Signal messages.  But I don't think it is appropriate for SMS.  Certainly it is not appropriate for an SMS received via a *different* SIM[1].  In particular, in a dual SIM phone, one wants to able to test a SIM by sending and receiving messages.  The behaviour of Signal confused me enough that I thought one of my telcos was at fault.

### Additional information

[1] IME Android does not seem able to reliably establish the phone numbers for each SIM.  For example, my device's "about phone" gives wrong information for the phone number associated with my main SIM, and doesn't even pretend to know the phone number associated with my aux SIM.

In my case the "main" SIM is an eSIM.  I recently had my main number ported to that SIM.  My aux SIM is a physical SIM whose setup I haven't changed in some time.

With the help of friend at the same telco as my aux sim, I was able to establish that my aux sim's telco *can* exchange SMSs with my main sim in both directions.  Additionally I have verified, by putting my aux SIM in another device, that the two SIMs are able to exchange SMS as expected.

### Device info

**Device:** Fairphone 4 (Murena, e-OS as shipped)
**Android version:** 11
**Signal version:** 5.38.5

### Link to debug log

https://debuglogs.org/android/5.38.5/60198a5fe7b60c98ebdcb9cd79bfac52eb39c27bf9b489131ebbc93730796fb0

**Comment by ijackson** [comment]
I still think it would be nice if this could be fixed.

I could go to the effort of checking whether it is still broken, but it doesn't seem worthwhile if all that will happen is for a robot to close the report anyway.

**Comment by PHCEAC** [comment]
I confirm this. I see in my debug log a message identical to those in issue #12428 
    "SmsReceiveJob : Received an SMS from ourselves! Ignoring."
Silently deleting SMS or other messages, even - or especially - those coming from one of our own telephone numbers, is a terrible mis-feature, and has broken several of my workflows. I've been blaming my mobile provider for ages.
See also the reference to the commit [c7a345eb0bb56026ad07fa7d66e872650b1521e6] which seems to make a fix clear.

**Comment by TBDwarf** [comment]
Yes, the same problem exists with me, my father and my wife as well as with colleagues.
As explained by PHCEAC, we also exchange on [#12428](https://github.com/signalapp/Signal-Android/issues/12428) 
All the users of the French telephone operator ''Free Mobile'' are technically confronted with the problem.

**Comment by Lineflyer** [comment]
I guess this is outdated as signal does no longer support SMS.

**Comment by ijackson** [comment]
> I guess this is outdated as signal does no longer support SMS.

Indeed.  Closing.  Thanks.
